### PR TITLE
feat: Added parameter to attach aws_eip

### PR DESCRIPTION
**Pull request by tokarev-artem**
## Description
Added parameter to create an attach aws_eip resource to ec2 instance

## Motivation and Context
It allows to attach public static IP address which will not be changed after ec2 instance stop/start

## How Has This Been Tested?
- [*] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [*] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [*] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: variables.tf
Comment (antonbabenko) [inline variables.tf:423]: Please swap lines 423 and 424

File: variables.tf
Comment (antonbabenko) [inline variables.tf:421]: Add an empty line between blocks.

**Review by antonbabenko — CHANGES_REQUESTED** [review]
LGTM, minor comments.

@bryantbiggs WDYT?

File: variables.tf
Comment (RealArtemiy) [reply variables.tf]: Done

File: variables.tf
Comment (RealArtemiy) [reply variables.tf:422]: Added

**Review by bryantbiggs — APPROVED** [review]
looks good to me - although, I am curious why you are using an EIP on a standalone instance in 2024 😅

File: main.tf
Comment (antonbabenko) [inline main.tf:619]: Please rename var to eip_domain

File: main.tf
Comment (antonbabenko) [inline main.tf:621]: And add another var with type map for eip_tags

**Review by antonbabenko — CHANGES_REQUESTED** [review]

File: main.tf
Comment (RealArtemiy) [reply main.tf]: renamed

File: main.tf
Comment (RealArtemiy) [reply main.tf]: Added

**Comment by tokarev-artem** [comment]
Hi @antonbabenko is there anything else required to add/change/edit?

**Comment by mikecook** [comment]
> looks good to me - although, I am curious why you are using an EIP on a standalone instance in 2024 😅

@bryantbiggs @antonbabenko I'd like this feature for a bastion server that doesn't need high availability.

**Comment by bryantbiggs** [comment]
> I'd like this feature for a bastion server for a reporting tool that doesn't need high availability.

And SSM session manager doesn't work for that? (I've approved the PR from my end, more of a pet peeve of this behavior over more modern practices really)

**Comment by mikecook** [comment]
> > I'd like this feature for a bastion server for a reporting tool that doesn't need high availability.
> 
> And SSM session manager doesn't work for that? (I've approved the PR from my end, more of a pet peeve of this behavior over more modern practices really)

I understand. No way to use SSM in this case, to my knowledge. I'm limited by 3rd party configuration options. The reporting tool is 3rd party automation that uses ssh-key to pass through bastion via SSH tunnel. It doesn't do anything on the ec2 instance.

**Comment by mikecook** [comment]
@antonbabenko Heads up, this PR is approved but is blocked by your acceptance of your requested changes.

**Comment by mikecook** [comment]
> @antonbabenko Heads up, this PR is approved but is blocked by your acceptance of your requested changes.

@antonbabenko This is ready. A user has done the work. Please respond to your blocking review feedback that has been addressed. This was marked stale while awaiting your response. It is otherwise ready to merge.

CC: @bryantbiggs

**Comment by mikecook** [comment]
@apparentlymart @shawnsavour @bryantbiggs Can anyone unblock this?

**Comment by bryantbiggs** [comment]
I cannot remove a review that requests changes - we'll need to wait for Anton to review and update

**Review by shawnsavour — APPROVED** [review]

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 5.7.0 :tada:

**Comment by carnei-ro** [comment]
i'm facing this problem now:

```
Error: Value for unconfigurable attribute

  with module.ec2_instance.aws_eip.this,
  on .terraform/modules/ec2_instance/main.tf line 619, in resource "aws_eip" "this":
 619:   domain = var.eip_domain

Can't configure a value for "domain": its value will be decided automatically
```

**Comment by bryantbiggs** [comment]
ugh, the EIP resource brought in a v5.0 argument https://github.com/hashicorp/terraform-provider-aws/pull/31567

Looks like we'll have to take a breaking change

**Pull request merged**
